components/script/dom/window.rs

```diff
@@ -555,6 +569,26 @@ impl Window {
         }
     }
 
+    pub(crate) fn pending_image_notification(&self, response: PendingImageResponse) {
+        let mut images = self.pending_image_callbacks.borrow_mut();
+        let Entry::Occupied(callbacks) = images.entry(response.id) else {
+            return;
+        };
+
+        for callback in callbacks.get() {
+            callback(self, response.clone());
```

Comment:
If we end up removing the task queueing, we'll have an outstanding mutable borrow here and callbacks that contain `CanGc::note()`.

Reply:
I have gone ahead and done a `std::mem::take` here for the duration of the method.